lib/util/git/index.ts

```diff
@@ -395,140 +397,155 @@ export function isCloned(): boolean {
-        throw err;
+      if (err.message.includes('does not have any commits yet')) {
+        throw new Error(REPOSITORY_EMPTY);
       }
-      throw new ExternalHostError(err, 'git');
+      logger.warn({ err }, 'Cannot retrieve latest commit');
     }
-    const durationMs = Math.round(Date.now() - cloneStart);
-    logger.debug({ durationMs }, 'git clone completed');
-  }
-  try {
-    config.currentBranchSha = (
-      await git.raw(['rev-parse', 'HEAD'])
-    ).trim() as LongCommitSha;
+    config.currentBranch =
+      config.currentBranch ??
+      config.defaultBranch ??
+      (await getDefaultBranch(git));
     /* v8 ignore next -- TODO: add test */
-  } catch (err) {
-    if (err.message?.includes('fatal: not a git repository')) {
-      throw new Error(REPOSITORY_CHANGED);
-    }
-    throw err;
-  }
-  // This will only happen now if set in global config
-  await cloneSubmodules(!!config.cloneSubmodules, config.cloneSubmodulesFilter);
-  try {
-    const latestCommit = (await git.log({ n: 1 })).latest;
-    logger.debug({ latestCommit }, 'latest repository commit');
-    /* v8 ignore next -- TODO: add test */
-  } catch (err) {
-    const errChecked = checkForPlatformFailure(err);
-    if (errChecked) {
-      throw errChecked;
-    }
-    if (err.message.includes('does not have any commits yet')) {
-      throw new Error(REPOSITORY_EMPTY);
-    }
-    logger.warn({ err }, 'Cannot retrieve latest commit');
-  }
-  config.currentBranch =
-    config.currentBranch ??
-    config.defaultBranch ??
-    (await getDefaultBranch(git));
-  /* v8 ignore next -- TODO: add test */
-  delete getCache()?.semanticCommits;
-
-  // If upstreamUrl is set then the bot is running in fork mode
-  // The "upstream" remote is the original repository which was forked from
-  if (config.upstreamUrl) {
-    logger.debug(
-      `Bringing default branch up-to-date with ${RENOVATE_FORK_UPSTREAM}, to get latest config`,
-    );
-    // Add remote if it does not exist
-    const remotes = await git.getRemotes(true);
-    if (!remotes.some((remote) => remote.name === RENOVATE_FORK_UPSTREAM)) {
-      logger.debug(`Adding remote ${RENOVATE_FORK_UPSTREAM}`);
-      await git.addRemote(RENOVATE_FORK_UPSTREAM, config.upstreamUrl);
+    delete getCache()?.semanticCommits;
+
+    // If upstreamUrl is set then the bot is running in fork mode
+    // The "upstream" remote is the original repository which was forked from
+    if (config.upstreamUrl) {
+      await instrument('sync with upstreamUrl', async () => {
+        logger.debug(
+          `Bringing default branch up-to-date with ${RENOVATE_FORK_UPSTREAM}, to get latest config`,
+        );
+        // Add remote if it does not exist
+        const remotes = await git.getRemotes(true);
+        if (!remotes.some((remote) => remote.name === RENOVATE_FORK_UPSTREAM)) {
+          logger.debug(`Adding remote ${RENOVATE_FORK_UPSTREAM}`);
+          await git.addRemote(RENOVATE_FORK_UPSTREAM, config.upstreamUrl!);
```

Comment:
Was needed now we're in the loop

Reply:
```suggestion
          await git.addRemote(RENOVATE_FORK_UPSTREAM, upstreamUrl);
```

This makes it more safe